internal/impl/kafka/franz_reader_unordered.go

```diff
@@ -560,9 +566,12 @@ func (f *FranzReaderUnordered) Connect(ctx context.Context) error {
 					}
 				}
 
-				if nonTemporalErr {
-					cl.Close()
-					return
+				if nonTemporalErr && fetches.Empty() {
+					select {
+					case <-time.After(connErrBackOff.NextBackOff()):
```

Comment:
don't we need to reset the backoff at some point?

Reply:
Whoops, you have good eye-meat, thanks